src/location.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from "react";
```

Comment:
Can we put this file under utils?

Reply:
Still a WIP file, file will probably not even exist in the next PR

Reply:
All guchi then